Lib/venv/__init__.py

```diff
@@ -164,7 +164,16 @@ def create_if_needed(d):
                              'Python interpreter. Provide an explicit path or '
                              'check that your PATH environment variable is '
                              'correctly set.')
-        dirname, exename = os.path.split(os.path.abspath(executable))
+        # only resolve executable symlinks, not the full chain, see gh-106045
+        # we don't want to overwrite the executable used in context
+        executable_ = os.path.abspath(executable)
+        while os.path.islink(executable_):
+            link = os.readlink(executable_)
```

Comment:
Technically we need to watch out for circular links. I think having this in a separate function (as suggested) will make that less painful when implementing.

FWIW, I don't know why there isn't something in the standard library for recursively resolving a symlink (without recursively resolving _all child paths_ as `realpath` does). Perhaps one for Python ideas?

Reply:
done